tempodb/backend/azure/compactor.go

```diff
@@ -127,33 +124,33 @@ func (rw *readerWriter) readAllWithModTime(ctx context.Context, name string) ([]
 	return bytes, att.LastModified, nil
 }
 
-// Attributes returns information about the specified blob using his name.
+// getAttributes returns information about the specified blob using its name.
 func (rw *readerWriter) getAttributes(ctx context.Context, name string) (BlobAttributes, error) {
-	blobURL, err := GetBlobURL(ctx, rw.cfg, name)
+	blobClient, err := getBlobClient(ctx, rw.cfg, name)
 	if err != nil {
-		return BlobAttributes{}, errors.Wrapf(err, "cannot get Azure blob URL, name: %s", name)
+		return BlobAttributes{}, errors.Wrapf(err, "cannot get Azure blob client, name: %s", name)
 	}
 
-	var props *blob.BlobGetPropertiesResponse
-	props, err = blobURL.GetProperties(ctx, blob.BlobAccessConditions{}, blob.ClientProvidedKeyOptions{})
+	props, err := blobClient.GetProperties(ctx, &blob.GetPropertiesOptions{})
 	if err != nil {
 		return BlobAttributes{}, err
 	}
 
 	return BlobAttributes{
-		Size:         props.ContentLength(),
-		LastModified: props.LastModified(),
+		Size:         *props.ContentLength,
+		LastModified: *props.LastModified,
 	}, nil
 }
 
 // Delete removes the blob with the given name.
 func (rw *readerWriter) delete(ctx context.Context, name string) error {
```

Comment:
is this different than the other `Delete()` method? should we consolidate?

Reply:
Looks like the two methods are mostly identical. The exported `Delete()` method returns `readError(err)` if `blobClient.Delete()` errors, but they are otherwise the same.

Action taken: removed `delete()` method in favour of `Delete()`.